R/render.r

```diff
@@ -422,7 +422,121 @@ data_deps <- function(pkg, depth) {
+        bs_theme,
+        "component-active-bg" = as.character(
+          bslib::bs_get_variables(bs_theme, "secondary")
+        )
+      )
+    }
+  } else {
+    if (is.null(pkg$meta[["template"]]$bslib$`component-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "component-active-bg" = "mix($body-color, $body-bg, 5%)",
+        .where = "declarations"
+      )
+    }
+    if (is.null(pkg$meta[["template"]]$bslib$`dropdown-link-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "dropdown-link-active-bg" = "mix($body-color, $body-bg, 5%)",
+        .where = "declarations"
+      )
+    }
+  }
+
+  # map body color to navbar colors
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-color" = "rgba($body-color, 0.8)",
+      .where = "declarations"
+    )
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-hover-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-hover-color" = "rgba($body-color, 0.9)",
+      .where = "declarations"
+    )
+  }
+
+  # map primary if set, or fg otherwise
+  # to navbar-light-active-color + component-active-color
+  # unless a value was set by the user
+  default_prim_color <- if (!is.null(pkg$meta[["template"]][["bslib"]][["primary"]])) {
+    pkg$meta[["template"]][["bslib"]][["primary"]]
+  } else {
+    as.character(bslib::bs_get_variables(bs_theme, "fg"))
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-active-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-active-color" = default_prim_color
+    )
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`component-active-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "component-active-color" = default_prim_color
+    )
+  }
+
+  # Map body-color to headings color
+  if (is.null(pkg$meta[["template"]]$bslib$`headings-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "headings-color" = as.character(bslib::bs_get_variables(bs_theme, "fg"))
+    )
+  }
+
+  # map component-active-color to dropdown-link-active-color
+  if (is.null(pkg$meta[["template"]]$bslib$`dropdown-link-active-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "dropdown-link-active-color" = "$component-active-color",
+      .where = "declarations"
+    )
+  }
+
+  # map primary, fg, bg to dropdown hover/focus
+  if (is.null(pkg$meta[["template"]]$bslib$`dropdown-link-hover-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "dropdown-link-hover-color" = default_prim_color
+    )
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`dropdown-link-hover-bg`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "dropdown-link-hover-bg" = "mix($body-color, $body-bg, 5%)",
+      .where = "declarations"
```

Comment:
Avoid the `.where = "declarations"` (see https://github.com/r-lib/pkgdown/pull/1682/files#r664795030)